common/core/nesteddisplay.cpp

```diff
@@ -67,10 +192,118 @@ bool NestedDisplay::SetPowerMode(uint32_t /*power_mode*/) {
   return true;
 }
 
-bool NestedDisplay::Present(std::vector<HwcLayer *> & /*source_layers*/,
+bool NestedDisplay::Present(std::vector<HwcLayer *> &source_layers,
                             int32_t * /*retire_fence*/,
                             bool /*handle_constraints*/) {
-  // TODO implement Present
+#ifdef NESTED_DISPLAY_SUPPORT
```

Comment:
We can limit this compiler check to line 229 to 238 ?

Reply:
I didn't get your point, could you describe it more clearly? you means the check start from
197   for (size_t layer_index = 0; layer_index < size; layer_index++) {
198     HwcLayer* layer = source_layers.at(layer_index);
199     if (!layer->IsVisible())
200       continue;

Reply:
I was thinking if this compiler check can be around only IOCTL calls ?

Reply:
I suggest not, because while no buffer sharing, the Present function do nothing, just return. so no need places some buffer sharing code